src/stk/molecular/molecules/building_block.py

```diff
@@ -644,15 +653,50 @@ def _normalize_placer_ids(self, placer_ids, functional_groups):
         """
 
         if placer_ids is not None:
-            return placer_ids
+            return frozenset(placer_ids)
 
         if functional_groups:
-            return tuple(flatten(
+            return frozenset(flatten(
                 functional_group.get_placer_ids()
                 for functional_group in functional_groups
             ))
 
-        return tuple(atom.get_id() for atom in self._atoms)
+        return frozenset(atom.get_id() for atom in self._atoms)
+
+    def _get_core_ids(
+        self,
+        functional_groups: Iterable[FunctionalGroup],
+    ) -> frozenset[int]:
```

Comment:
Tbh, maybe making this a generator would be better, where the generator will return duplicates. My reasoning being that you can then call it like this

```python
self._core_ids = frozenset(self._get_core_ids(...))
```

Where its immediately obvious that you re deduping.  With

```python
self._core_ids = self._get_core_ids(...)
```

The reader has to jump to the function definition to see the frozenset.

You'll have to add a comment saying that it might return dupes though.